File: v0/server/src/Application/TeamBarometer/Models/MeetingModel.cs
using Domain.TeamBarometer.Entities;
using System;
using System.Collections.Generic;
using System.Linq;

namespace Application.TeamBarometer.Models
{
	public class MeetingModel
	{
		public MeetingModel(Meeting meeting, Guid userId)
		{
			Id = meeting.Id;
			Questions = meeting.Questions.Select(question => new QuestionModel(question));
			UserIsTheFacilitator = meeting.UserIsTheFacilitator(userId);
		}

		public Guid Id { get; set; }
		public IEnumerable<QuestionModel> Questions { get; private set; }
		public bool UserIsTheFacilitator { get; set; }

		public override bool Equals(object obj)
		{
			MeetingModel meeting = obj as MeetingModel;

			return Id == meeting?.Id;
		}

		public override int GetHashCode()
		{
			return Id.GetHashCode();
		}
	}
}

File: v0/client/react/src/meeting/models/MeetingModel.js
import QuestionModel from "./QuestionModel";

export default class MeetingModel {
    constructor({id, questions, userIsTheFacilitator}) {
        this.id = id;
        this.questions = (questions || []).map(question => new QuestionModel(question));
        this.userIsTheFacilitator = userIsTheFacilitator;
    }

    getCurrentQuestion() {
        let currentQuestion = this.questions.find(q => q.isTheCurrent);

        if (!currentQuestion) {
            currentQuestion = this.getLastQuestion();
        }

        return currentQuestion;
    }

    getLastQuestion() {
        return this.questions[this.questions.length - 1];
    }
}